Converters/Rhino/Speckle.Converters.RhinoShared/ToSpeckle/Meshing/DisplayMeshExtractor.cs

```diff
@@ -76,4 +82,77 @@ public static RG.Mesh GetDisplayMesh(RhinoObject obj)
     joinedMesh.Append(renderMeshes);
     return joinedMesh;
   }
+
+  /// <summary>
+  /// Calculating optimal meshing parameter 'minimumEdgeLength' for the given geometry.
+  /// </summary>
+  private static double GetAccurateMinEdgeLegth(RG.GeometryBase geometry)
+  {
+    // adjust meshing parameters if Brep edges are too close to the document tolerance
+    double minEdgeLength = 0.05;
+    if (geometry is RG.Brep brep && brep.Edges.Any(x => x.GetLength() < minEdgeLength))
+    {
+      return 0;
+    }
+
+    return minEdgeLength;
+  }
+
+  /// <summary>
+  /// Extracting Rhino Mesh from Rhino GeometryBase after moving it to origin (if needed).
+  /// </summary>
+  public static RG.Mesh MoveToOriginAndGetDisplayMesh(
+    RG.GeometryBase geometry,
+    bool modelFarFromOrigin,
+    out RG.Vector3d? vectorToOriginalGeometry
+  )
+  {
+    vectorToOriginalGeometry = null;
+
+    // 1. General check: if Model is NOT far from origin (99% of Rhino models): extract meshes as usual
+    if (!modelFarFromOrigin)
+    {
+      return geometry is RG.Mesh mesh ? mesh : GetGeometryDisplayMesh(geometry, true);
+    }
+    // 2. Geometry check: if the model extent is far from origin, but object itself is NOT far from origin: extract meshes as usual
+    if (!TryGetTranslationVector(geometry, out RG.Vector3d vectorToGeometry))
+    {
+      return geometry is RG.Mesh mesh ? mesh : GetGeometryDisplayMesh(geometry, true);
+    }
+    // 3. If the object is far from origin and risking faulty meshes due to precision errors: duplicate geometry and move it to origin
+    RG.GeometryBase geometryToMesh = geometry.Duplicate();
+    geometryToMesh.Transform(RG.Transform.Translation(-vectorToGeometry));
+
+    vectorToOriginalGeometry = vectorToGeometry;
+    return geometryToMesh is RG.Mesh movedMesh ? movedMesh : GetGeometryDisplayMesh(geometryToMesh, true);
+  }
+
+  /// <summary>
+  /// Getting translation vector from origin to the Geometry bbox Center (if geometry is far from origin and translation needed)
+  /// </summary>
+  /// <returns>
+  /// True and the vector from origin to Geometry bbox center (if translation needed), otherwise false and zero-length vector.
+  /// </returns>
+  private static bool TryGetTranslationVector(RG.GeometryBase geom, out RG.Vector3d vector)
```

Comment:
This is misleading, tryget implies there's some failure to retrieve translation vector, it does not indicate not needing to retrieve translation vector. In any case, I would just have this be the output of the distance check extension method, like so:
```
public bool IsFarFromOrigin(this GeometryBase geo, out Vector translationVector) { 
// test for distance and set bool
// set translation vector default to 0, calculate vector if bool = true
}